Readme.md

```diff
@@ -1,155 +1,135 @@
+$ git clone https://github.com/rust-lang/gcc gcc-source
 $ sudo apt install flex libmpfr-dev libgmp-dev libmpc3 libmpc-dev
 $ mkdir gcc-build gcc-install
 $ cd gcc-build
-$ ../gcc/configure \
+$ ../gcc-source/configure \
     --enable-host-shared \
     --enable-languages=jit \
-    --enable-checking=release \ # it enables extra checks which allow to find bugs
+    --enable-checking=release \
     --disable-bootstrap \
     --disable-multilib \
     --prefix=$(pwd)/../gcc-install
-$ make -j4 # You can replace `4` with another number depending on how many cores you have.
+$ make -j4
+$ make install
 ```
 
-If you want to run libgccjit tests, you will need to also enable the C++ language in the `configure`:
+Notes:
+* If you want to run libgccjit tests, you must also add C++ to the enabled languages: `--enable-languages=jit,c++`
+* `--enable-host-shared` builds the compiler as position independent code and is required to build libgccjit. This results in a slower compiler; however, if building GCC in multiple passes, `jit` may be built in the first pass with `--enable-host-shared`, with both disabled in subsequent passes.
+* `--enable-checking=release` and `--disable-bootstrap` speed up compilation by disabling self-checks and may be omitted.
+* `--disable-multilib` is used as libgccjit only supports targeting one arch ABI variant.
+* Adjust the `4` in `make -j4` to the number of threads available on your system to speed up compilation.
+* Change `make install` to `make install-strip` to remove debug symbols from GCC, including `libgccjit.so`, to save on disk space.
 
+Once the build is complete, one may run libgccjit tests like so (requires DejaGnu to be installed):
 ```bash
---enable-languages=jit,c++
+$ make -C gcc check-jit
 ```
-
-Then to run libgccjit tests:
-
+To run one specific test:
 ```bash
-$ cd gcc # from the `gcc-build` folder
-$ make check-jit
-# To run one specific test:
-$ make check-jit RUNTESTFLAGS="-v -v -v jit.exp=jit.dg/test-asm.cc"
+$ make -C gcc check-jit RUNTESTFLAGS="-v -v -v jit.exp=jit.dg/test-asm.cc"
 ```
 
-**Put the path to your custom build of libgccjit in the file `config.toml`.**
-
-You now need to set the `gcc-path` value in `config.toml` with the result of this command:
-
+Now that you've compiled GCC with libgccjit support, the installed GCC toolchain with `libgccjit.so` can be found in the `gcc-install` directory created above. You will need to provide the path to the specific subdirectory containing `libgccjit.so` to rustc_codegen_gcc. Use the following commands to leave the `gcc-build` directory and find the absolute path within the `gcc-install` directory:
 ```bash
-$ dirname $(readlink -f `find . -name libgccjit.so`)
+$ cd ..
+$ dirname $(readlink -f `find ./gcc-install -name libgccjit.so`)
 ```
 
-and to comment the `download-gccjit` setting:
+If desired, you may now delete the `gcc-source` and `gcc-build` directories to reclaim disk space, but keep the `gcc-install` directory.
```

Comment:
I wouldn't recommend that.

Reply:
I guess that may be useful for people wanting the use the project, but not contribute on it.